apps/web/app/(app)/environments/[environmentId]/settings/billing/actions.ts

```diff
@@ -0,0 +1,132 @@
+"use server";
+
+import { authOptions } from "@formbricks/lib/authOptions";
+import { getServerSession } from "next-auth";
+import { AuthorizationError } from "@formbricks/types/errors";
+import { WEBAPP_URL } from "@formbricks/lib/constants";
+import { canUserAccessTeam } from "@formbricks/lib/team/auth";
+import { getTeam } from "@formbricks/lib/team/service";
+import { getMonthlyActivePeopleCount } from "@formbricks/lib/person/service";
+import { getMonthlyDisplayCount } from "@formbricks/lib/display/service";
+
+export async function getBillingDetails(teamId: string) {
+  const session = await getServerSession(authOptions);
+  if (!session) throw new AuthorizationError("Not authorized");
+
+  const isAuthorized = await canUserAccessTeam(session.user.id, teamId);
+  if (!isAuthorized) throw new AuthorizationError("Not authorized");
+
+  const team: any = await getTeam(teamId);
+
+  const stripeCustomerId = team.subscription?.stripeCustomerId;
+
+  if (!stripeCustomerId) {
+    return {
+      people: -1,
+      display: -1,
+    };
+  }
+
+  let peopleForTeam = 0;
+  let displaysForTeam = 0;
+
+  for (const product of team.products) {
+    for (const environment of product.environments) {
+      const peopleInThisEnvironment = await getMonthlyActivePeopleCount(environment.id);
+      const displaysInThisEnvironment = await getMonthlyDisplayCount(environment.id);
+
+      peopleForTeam += peopleInThisEnvironment;
+      displaysForTeam += displaysInThisEnvironment;
+    }
+  }
+
+  return {
+    people: peopleForTeam,
+    display: displaysForTeam,
+  };
+}
+
+export async function upgradePlanAction(teamId: string, environmentId: string, priceNickname: string) {
+  const session = await getServerSession(authOptions);
+  if (!session) throw new AuthorizationError("Not authorized");
+
+  const isAuthorized = await canUserAccessTeam(session.user.id, teamId);
+  if (!isAuthorized) throw new AuthorizationError("Not authorized");
+
+  const res = await fetch(WEBAPP_URL + "/api/billing/create-subscription", {
```

Comment:
why do we have this as an api call. since we are already on the server we could just call the `createSubscription` method?

Reply:
I had API calls everywhere under the assumption that we want to keep all the `ee/` package abstractions away from the app! Will make all these to use direct methods!

Reply:
That's right, but we can also call a function from the ee package. The only important thing is that the code is in the ee package so it falls under the ee license :-)